.github/workflows/test-builds-linux.yml

```diff
@@ -0,0 +1,102 @@
+name: Linux Test Builds
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
```

Comment:
It seems like this is not working, at least I have not way of "approving" this workflow.

Reply:
After doing some testing on my fork, it looks like the workflow will not be run if the workflow is not defined on the target branch of a PR. Once the workflow is defined in the target branch, any changes made by a future PR will effect the workflow during those future PRs.

e.g. I have the branches `$default-branch` (`master` in this case) and `github_ci_test` in the allowed list of branches for the `pull_request` event. If I open a PR of `github_ci_test` onto `master`, the workflow will not be run. If I open a PR of `github_ci_test_test` onto `github_ci_test` with some random code change, the workflow is run. If I open a PR of `github_ci_test_test` onto `github_ci_test` with a change removing `github_ci_test` from the allow list, the workflow does not run.

Once again I seem to have run into a case where I don't have a good way for you folks to test my changes at the moment. I have found https://github.com/nektos/act which allows you to run a workflow locally, but this is not ideal. I will continue looking into this.

A slight side note, I can't say I'm very happy with this "feature". I get that it is a security feature to prevent random PRs form untrusted people from leaking secrets, but it makes it a pain to switch over to. I can't say I have ever run into anything like this on another platform.

Reply:
Looking at other projects, I think the path forward here is to add GitHub Actions as a CI path without removing Travis. Afterwords, if the workflow is working as expected the old Travis configuration can be removed.

Reply:
Good idea. Would it be possible for you to run the latest version of these builds on your forked repo for us to see the results?

Reply:
Here is an example PR to show the results.

https://github.com/yodaldevoid/libssh2/pull/5